fili-core/src/test/groovy/com/yahoo/bard/webservice/filterbuilders/DruidBoundFilterBuilderSpec.groovy

```diff
@@ -0,0 +1,81 @@
+import com.yahoo.bard.webservice.data.filterbuilders.DruidFilterBuilder
+import com.yahoo.bard.webservice.druid.model.filter.BoundFilter
+import com.yahoo.bard.webservice.druid.model.filter.Filter
+import com.yahoo.bard.webservice.web.apirequest.binders.FilterBinders
+
+import spock.lang.Shared
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class DruidBoundFilterBuilderSpec extends Specification {
+    @Shared QueryBuildingTestingResources resources
+    Map filterSpecs
+    Map apiFilters
+    Map druidFilters
+    DruidFilterBuilder filterBuilder
+    FilterBinders filterBinders = FilterBinders.INSTANCE
+
+    def setupSpec() {
+        resources = new QueryBuildingTestingResources()
+    }
+
+    def setup() {
+        filterSpecs = [
+                startDateIdLt        : "dim7|id-lt[2018-08-10]",
+                startDateIdGte       : "dim7|id-gte[2016-08-10]",
+                startDateIdBetween   : "dim7|id-between[2016-08-10,2018-08-10]"
+        ]
+        apiFilters = [:]
+        filterSpecs.each {
+            apiFilters.put(it.key, filterBinders.generateApiFilter(it.value, resources.dimensionDictionary))
+        }
+
+        Filter startDateLtBoundFilter = new BoundFilter(resources.d7, null, "2018-08-10").withUpperBoundStrict(true)
+        Filter startDateGteBoundFilter = new BoundFilter(resources.d7, "2016-08-10", null)
+        Filter startDateBetweenBoundFilter = new BoundFilter(resources.d7, "2016-08-10", "2018-08-10").withUpperBoundStrict(true)
+
+        druidFilters = [
+                startDateIdLt        : startDateLtBoundFilter,
+                startDateIdGte       : startDateGteBoundFilter,
+                startDateIdBetween   : startDateBetweenBoundFilter
+        ]
+
+        filterBuilder = new DruidBoundFilterBuilder()
+    }
+
+    @Unroll
+    def "Filtering for #apiList on #apiName gives us a Druid filter of type #filterType on #druidList"() {
+        setup:
+        LinkedHashSet apiSet = apiList.collect { apiFilters.get(it) } as LinkedHashSet
+
+        List druidExpected = druidList.collect() { druidFilters.get(it) }
+
+        when:
+        Filter outerFilter = filterBuilder.buildFilters([(resources.d7): apiSet])
+        List<Filter> filters = outerFilter.type == AND ? outerFilter.fields : [outerFilter]
+
+        then:
+        outerFilter.type == filterType
+        if (outerFilter.type == BOUND ) {
+            assert druidExpected == [outerFilter]
+        } else {
+            assert filters.containsAll(druidExpected) && druidExpected.containsAll(filters)
+        }
+
+        where:
+        apiName = resources.d7.apiName
+        apiList                               | filterType | druidList
+        ["startDateIdLt"]                     | BOUND      | ["startDateIdLt"]
+        ["startDateIdLt", "startDateIdGte"]   | AND        | ["startDateIdLt", "startDateIdGte"]
+        ["startDateIdBetween"]                | BOUND      | ["startDateIdBetween"]
+    }
```

Comment:
If we throw exception on multi-dimension values, we should also add a test for it